projects/kit/components/slider/slider.component.ts

```diff
@@ -1,84 +1,93 @@
+    selector: 'input[tuiSlider]',
+    template: ``,
+    styleUrls: ['./slider.style.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [
-        {
-            provide: TUI_FOCUSABLE_ITEM_ACCESSOR,
-            useExisting: forwardRef(() => TuiSliderComponent),
-        },
-    ],
 })
-export class TuiSliderComponent
-    extends AbstractTuiSlider<number>
-    implements TuiFocusableElementAccessor
-{
-    constructor(
-        @Optional()
-        @Self()
-        @Inject(NgControl)
-        control: NgControl | null,
-        @Inject(ChangeDetectorRef) changeDetectorRef: ChangeDetectorRef,
-        @Inject(DOCUMENT) documentRef: Document,
-        @Inject(TUI_FROM_TO_TEXTS) fromToTexts$: Observable<[string, string]>,
-    ) {
-        super(control, changeDetectorRef, documentRef, fromToTexts$);
-    }
+export class TuiSliderComponent {
+    @Input()
+    @HostBinding('attr.data-size')
+    @tuiDefaultProp()
+    size: TuiSizeS = 'm';
 
-    get nativeFocusableElement(): TuiNativeFocusableElement | null {
-        return this.dotRight ? this.dotRight.nativeElement : null;
-    }
+    @Input()
+    @tuiDefaultProp()
+    @HostBinding('class._ticks')
+    showTicks = false;
 
-    get focused(): boolean {
-        return isNativeFocused(this.nativeFocusableElement);
+    @HostBinding('attr.type')
+    readonly type = 'range';
+
+    get min(): number {
+        return +this.elementRef.nativeElement.min;
     }
 
-    get left(): number {
-        return 0;
+    get max(): number {
+        return +this.elementRef.nativeElement.max || 100;
     }
 
-    get right(): number {
-        return 100 - 100 * this.getFractionFromValue(this.value);
+    get step(): number {
+        return +this.elementRef.nativeElement.step || 1;
     }
 
-    protected getFallbackValue(): number {
-        return 0;
+    @HostBinding('style.--tui-slider-fill-percentage')
+    get fillPercentage(): string {
+        const percentage = Math.floor((100 * this.value) / (this.max - this.min));
+
+        return `${percentage}%`;
     }
 
-    protected processStep(increment: boolean) {
-        const fraction = this.getFractionFromValue(this.value);
-        const step = this.computedStep;
-        const value = this.getValueFromFraction(
-            increment ? fraction + step : fraction - step,
-        );
+    @HostBinding('style.--tui-slider-step-percentage')
+    get stepPercentage(): string {
+        const percentage = 100 / ((this.max - this.min) / this.step) - 0.1;
 
-        this.processValue(value);
+        return `${percentage}%`;
     }
 
-    protected processValue(value: number) {
-        this.updateValue(this.valueGuard(value));
+    @HostBinding('class._old-edge')
+    get isOldEdge(): boolean {
+        return isEdgeOlderThan(CHROMIUM_EDGE_START_VERSION, this.userAgent);
     }
+
+    get value(): number {
+        return +this.control?.value || +this.elementRef.nativeElement.value || 0;
```

Comment:
I think we do not need to convert control value and expect it to be number, like we usually do.

Reply:
fixed